src/plugins/EnergyRAPL.cpp

```diff
@@ -65,16 +70,43 @@ int EnergyRAPL::getNumberOfPackages() {
 
 void EnergyRAPL::init(ParticleContainer* particleContainer, DomainDecompBase* domainDecomp, Domain* domain) {
 #ifdef ENABLE_MPI
+	bool thisRankShouldMeasure = false;
+	MPI_Status status;
+	char processorName[MPI_MAX_PROCESSOR_NAME];
 	int processorNameLength;
-	MPI_Get_processor_name(_processorName, &processorNameLength);
+	MPI_Get_processor_name(processorName, &processorNameLength);
 	MPI_Comm_rank(MPI_COMM_WORLD, &_thisRank);
-#endif
-	if (!_outputprefix.empty()) {
-		std::ostringstream outputFilename;
-		outputFilename << _outputprefix << ".tsv";
-		std::ofstream outputFile(outputFilename.str().c_str());
-		outputFile << "milliseconds\tsimstep\tjoules" << std::endl;
+	if (_thisRank == 0) {
+		std::map<std::string, int> processorsRaplRank;
+		int numberOfRanks;
+		MPI_Comm_size(MPI_COMM_WORLD, &numberOfRanks);
+		for (int otherRank = 1; otherRank < numberOfRanks; otherRank++) {
+			char otherRankProcessorName[MPI_MAX_PROCESSOR_NAME];
+			MPI_Recv(otherRankProcessorName, MPI_MAX_PROCESSOR_NAME, MPI_CHAR, otherRank, /* tag */ otherRank, MPI_COMM_WORLD,
+					 &status);
+			processorsRaplRank[otherRankProcessorName] = otherRank;
+		}
+		processorsRaplRank[processorName] = _thisRank;  // Root rank should measure
+		std::set<int> raplRanks;                        // Only measure on one rank per processor (node)
+		for (auto item : processorsRaplRank) {
+			Log::global_log->debug() << "[" << getPluginName() << "] RAPL measurements for " << item.first
+									 << " are performed on MPI rank " << item.second << std::endl;
+			raplRanks.insert(item.second);
+		}
+		thisRankShouldMeasure = raplRanks.find(_thisRank) != raplRanks.end();
+		for (int i = 1; i < numberOfRanks; i++) {
+			bool otherRankShouldMeasure = raplRanks.find(i) != raplRanks.end();
+			MPI_Send(&otherRankShouldMeasure, 1, MPI_C_BOOL, /* dest */ i, /* tag */ i, MPI_COMM_WORLD);
```

Comment:
Nitpicking here ... MPI_C_BOOL corresponds to the C type `_Bool`. The C++ `bool` type might be different - or at least I do not know about anything in the C++ standard stating that `bool` is compatible with `_Bool`. However, most compilers will try to make things happen ... 
```suggestion
			MPI_Send(&otherRankShouldMeasure, 1, MPI_CXX_BOOL, /* dest */ i, /* tag */ i, MPI_COMM_WORLD);
```